src/common/image/extract_color.ts

```diff
@@ -0,0 +1,139 @@
+import Vibrant from "node-vibrant/lib/browser";
+import MMCQ from "@vibrant/quantizer-mmcq";
+import { BasicPipeline } from "@vibrant/core/lib/pipeline";
+import { Swatch, Vec3 } from "@vibrant/color";
+import { contrast } from "../color/contrast";
+
+const CONTRAST_RATIO = 4.5;
+
+// How much the total diff between 2 RGB colors can be
+// to be considered similar.
+const COLOR_SIMILARITY_THRESHOLD = 150;
+
+// For debug purposes, is being tree shaken.
+const DEBUG_COLOR = __DEV__ && false;
+
+function getContrastRatio(
+  rgb1: [number, number, number],
+  rgb2: [number, number, number]
+): number {
+  return Math.round((contrast(rgb1, rgb2) + Number.EPSILON) * 100) / 100;
+}
```

Comment:
We also have this: <https://github.com/home-assistant/frontend/blob/dev/src/common/color/rgb.ts#L12>

Reply:
So the contrast and luminance helpers could be removed too.